cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -40,6 +42,100 @@ pub struct ContractAliasInput {
     pub new_alias: Option<String>,
 }
 
+pub fn propose_pfm(
+    ctx: (DepsMut, Env, MessageInfo),
+    chain: String,
+) -> Result<Response, ContractError> {
+    let (deps, env, info) = ctx;
+
+    // enforce lowercase
+    let chain = chain.to_lowercase();
+
+    // validation
+    let registry = Registry::default(deps.as_ref());
+    let coin = cw_utils::one_coin(&info)?;
+    let native_chain = registry.get_native_chain(&coin.denom)?;
+
+    if native_chain.as_ref() != chain {
+        return Err(ContractError::CoinFronInvalidChain {
+            supplied_chain: native_chain.as_ref().to_string(),
+            expected_chain: chain,
+        });
+    }
+
+    // check if the chain is already registered or is in progress
+    if let Some(chain_pfm) = CHAIN_PFM_MAP.may_load(deps.storage, &chain)? {
+        if chain_pfm.is_validated() {
+            // Only authorized addresses can ask for a validated PFM to be re-checked
+            // If sender is the contract governor, then they are authorized to do do this to any chain
+            // Otherwise, they must be authorized to do manage the chain they are attempting to modify
+            let user_permission =
+                check_is_authorized(deps.as_ref(), info.sender, Some(chain.clone()))?;
+            check_action_permission(FullOperation::Change, user_permission)?;
```

Comment:
Asking to learn: what is the reason it works this way?

Reply:
This is to be compatible with how the permissions for the other registries work. Here the action is always Change, but there the action is passed in the message.

The reason only the contract governor, or someone authorized to manage the chain can change this is that changing this value once packaged are already flowing through this contract would DOS them, so we can't let anyone execute this.

In this case, the denial of service would only be temporary (while the packet is in-flight) but someone could keep proposing PFM checks and keep the validation in progress. This ensures only authorized parties can do that.